azure/durable_functions/models/actions/CompoundAction.py

```diff
@@ -31,5 +31,5 @@ def to_json(self) -> Dict[str, Union[str, int]]:
         """
         json_dict: Dict[str, Union[str, int]] = {}
         add_attrib(json_dict, self, 'action_type', 'actionType')
-        add_attrib(json_dict, self, 'compound_actions', 'compoundActions')
+        json_dict['compoundActions'] = list(map(lambda x: x.to_json(), self.compoundTasks))
```

Comment:
It seems there are a couple inconsistencies here and I'm wondering whether they were intentional:
1. In line 33 you use `add_attrib` whereas in line 34 you do a direct assignment. Are these two techniques equivalent?
2. It looks as though you are changing the field name from `compound_actions` (snake case) to `compoundActions` (camel case). Is that what's actually happening or does the serialized name differ depending on my previous point (`add_attrib` vs. direct `json_dict` assignment)?

Reply:
Actually, looking at this again, maybe (2) isn't really a thing. I see we have both the snake_case and the camelCase represented in the `add_attrib` function call.

Reply:
(1) Yes, they're equivalent.  To be frank, the `add_attrib` approach is a utility we inherited from the original authors, and in my mind it unnecessarily obfuscates what's really going on. In a different PR, I'd love to remove its usage for direct assignment.

(2) The camelCase is the name of the field in JSON. We serialize to camelCase because that's what the extension, in C#, expects in order to de-serialize WhenAll and WhenAny actions :). 

You're right though, I did make a styling error with `self.compoundTasks`, that should be `self.compound_tasks`. Let me fix that shortly